emails/views.py

```diff
@@ -1568,10 +1600,107 @@ def _handle_bounce(message_json: AWS_SNSMessageJSON) -> HttpResponse:
     return HttpResponse("OK", status=200)
 
 
+def _build_disabled_mask_for_spam_email(
+    mask: RelayAddress | DomainAddress, original_spam_email: dict
+) -> EmailMessage:
+    ctx = {
+        "mask": mask.full_address,
+        "spam_email": original_spam_email,
+        "SITE_ORIGIN": settings.SITE_ORIGIN,
+    }
+    html_body = render_to_string("emails/disabled_mask_for_spam.html", ctx)
+    text_body = render_to_string("emails/disabled_mask_for_spam.txt", ctx)
+
+    # Create the message
+    msg = EmailMessage()
+    msg["Subject"] = ftl_bundle.format("relay-disabled-your-mask")
+    msg["From"] = settings.RELAY_FROM_ADDRESS
+    msg["To"] = mask.user.email
+    msg.set_content(text_body)
+    msg.add_alternative(html_body, subtype="html")
+    return msg
+
+
+def _send_disabled_mask_for_spam_email(
+    mask: RelayAddress | DomainAddress, original_spam_email: dict
+) -> None:
+    msg = _build_disabled_mask_for_spam_email(mask, original_spam_email)
+    if not settings.RELAY_FROM_ADDRESS:
+        raise ValueError(
+            "Must set settings.RELAY_FROM_ADDRESS to send disabled_mask_for_spam email."
+        )
+    try:
+        ses_send_raw_email(
+            source_address=settings.RELAY_FROM_ADDRESS,
+            destination_address=mask.user.email,
+            message=msg,
+        )
+    except ClientError as e:
+        logger.error("reply_not_allowed_ses_client_error", extra=e.response["Error"])
+    incr_if_enabled("free_user_reply_attempt", 1)
+
+
+def _disable_masks_for_complaint(message_json: dict, user: User) -> None:
+    """
+    See https://docs.aws.amazon.com/ses/latest/dg/notification-contents.html#mail-object
+
+    message_json.mail.source contains the envelope MAIL FROM address from which the
+    original message was sent.
+
+    Relay sends emails From: "original-From" <relay-mask@mozmail.com>.
+
+    So, we can find the mask that sent the spam email by parsing the source value.
+    """
+    flag_name = "disable_mask_on_complaint"
+    _, _ = get_waffle_flag_model().objects.get_or_create(
+        name=flag_name,
+        defaults={
+            "note": (
+                "MPP-3119: When a Relay user marks an email as spam, disable the mask."
+            )
+        },
+    )
+    source = message_json.get("mail", {}).get("source", "")
+    # parseaddr is confused by 2 email addresses in the value, so use this
+    # regular expression to extract the mask address by searching for any relay domains
+    email_domains = get_domains_from_settings().values()
+    domain_pattern = "|".join(re.escape(domain) for domain in email_domains)
+    email_regex = rf"[\w\.-]+@(?:{domain_pattern})"
+    matches = re.findall(email_regex, source)
+    if not matches:
+        return
+    for mask_address in matches:
+        try:
+            address = _get_address(mask_address, False)
+            # ensure the mask belongs to the user for whom Relay received a complaint
+            if address.user != user:
```

Comment:
```suggestion
            if address.user != user or address.enabled == False:
```
(suggestion, non-blocking) Should we send the disabled mask email if the mask was already set to blocked? This can happen when when complaints was filed after the mask was already set to block by us or by the user.